src/acoupi/programs/templates/detection.py

```diff
@@ -189,6 +189,39 @@ def setup(self, config: C) -> None:
         self.model = self.configure_model(config)
         super().setup(config)
 
+    def on_end(self, deployment: data.Deployment) -> None:
+        """Handle program end event.
+
+        This method is called when the program ends. It updates the
+        deployment information in the metadata store, and ensure that
+        remaining tasks are completed before the program is stopped.
+
+        Tasks to check are:
+        - file_management_task (if implemented). Check if there are remaining
+        files in the temporary directory and move them to the correct directory.
+        - detection_task (if implemented). Check if there are remaining files
+        to be processed and process them.
+        """
+        super().on_end(deployment)
```

Comment:
Maybe it would be best to run the `super.on_end` in the end. This way the inherited `file_management_task` is executed after running the detections.

Reply:
I put it at the beginning as such it won't have to run the detection on files that have potentially already been processed. It will first remove the already processed files and then go through detection, file_management, and message for one last time.